src/base/exceptions.h

```diff
@@ -28,12 +28,30 @@
 
 #pragma once
 
-#include <stdexcept>
+#include <exception>
 #include <QString>
 
-class RuntimeError : public std::runtime_error
+class Exception : public std::exception
 {
 public:
-    explicit RuntimeError(const QString &message = {});
-    QString message() const;
+    explicit Exception(const QString &message = {});
+
+    [[nodiscard]] QString message() const noexcept;
+    [[nodiscard]] const char *what() const noexcept override;
+
+private:
+    QString m_message;
+    QByteArray m_utf8Data;
```

Comment:
where does it get the value?

Reply:
Oops... I missed it in the next rework. It must be assigned in constructor.

Reply:
Fixed.